velox/functions/prestosql/tests/ArrayPositionTest.cpp

```diff
@@ -0,0 +1,441 @@
+      {true, false},
+      {true},
+      {false},
+      {},
+      {std::nullopt},
+      {true, std::nullopt, true},
+      {false, false, false},
+  });
+
+  auto testPosition = [&](std::optional<bool> search,
+                          const std::vector<std::optional<int64_t>>& expected) {
+    auto result = evaluate<SimpleVector<int64_t>>(
+        "array_position(c0, c1)",
+        makeRowVector({
+            arrayVector,
+            makeConstant(search, arrayVector->size()),
+        }));
+
+    assertEqualVectors(makeNullableFlatVector<int64_t>(expected), result);
+  };
+
+  testPosition(true, {1, 1, 0, 0, 0, 1, 0});
+  testPosition(false, {2, 0, 1, 0, 0, 0, 1});
+  testPosition(
+      std::nullopt,
+      {std::nullopt,
+       std::nullopt,
+       std::nullopt,
+       std::nullopt,
+       std::nullopt,
+       std::nullopt,
+       std::nullopt});
+}
+
+TEST_F(ArrayPositionTest, row) {
+  std::vector<std::vector<variant>> data = {
+      {
+          variant::row({1, "red"}),
+          variant::row({2, "blue"}),
+          variant::row({3, "green"}),
+      },
+      {
+          variant::row({2, "blue"}),
+          variant(TypeKind::ROW), // null
+          variant::row({5, "green"}),
+      },
+      {},
+      {
+          variant(TypeKind::ROW), // null
+      },
+      {
+          variant::row({1, "yellow"}),
+          variant::row({2, "blue"}),
+          variant::row({4, "green"}),
+          variant::row({5, "purple"}),
+      },
+  };
+
+  auto rowType = ROW({INTEGER(), VARCHAR()});
+  auto arrayVector = makeArrayOfRowVector(rowType, data);
+
+  auto testPosition = [&](int32_t n,
+                          const char* color,
+                          const std::vector<std::optional<int64_t>>& expected) {
+    auto search =
+        makeConstantRow(rowType, variant::row({n, color}), arrayVector->size());
+
+    auto result = evaluate<SimpleVector<int64_t>>(
+        "array_position(c0, c1)", makeRowVector({arrayVector, search}));
+
+    assertEqualVectors(makeNullableFlatVector<int64_t>(expected), result);
+  };
+
+  testPosition(1, "red", {1, 0, 0, 0, 0});
+  testPosition(2, "blue", {2, 1, 0, 0, 2});
+  testPosition(4, "green", {0, 0, 0, 0, 3});
+  testPosition(5, "green", {0, 3, 0, 0, 0});
+  testPosition(1, "purple", {0, 0, 0, 0, 0});
+}
+
+TEST_F(ArrayPositionTest, integerWithInstance) {
+  auto arrayVector = makeNullableArrayVector<int64_t>(
+      {{1, 2, std::nullopt, 4},
+       {3, 4, 4},
+       {},
+       {std::nullopt},
+       {5, 2, 4, 2, 9},
+       {7},
+       {10, 4, 8, 4}});
+
+  auto testPosition = [&](std::optional<int64_t> search,
+                          std::optional<int64_t> instance,
+                          const std::vector<std::optional<int64_t>>& expected) {
+    auto result = evaluate<SimpleVector<int64_t>>(
+        "array_position(c0, c1, c2)",
+        makeRowVector(
+            {arrayVector,
+             makeConstant(search, arrayVector->size()),
```

Comment:
It appears as all tests use constant vectors for "search" and "instance" arguments. Is this so? 

It would be nice to test encodings for all types. There is an example of doing this generically in testExpr in velox/functions/prestosql/tests/ArrayIntersectTest.cpp

Reply:
I've made `search` and `instance` vectors both be constant vectors in dictionary encoding. In `TEST_F(ArrayPositionTest, array)` and `TEST_F(ArrayPositionTest, map)`, I've also tested with `search` being non-constant vectors. Please let me know if there are other encodings I should test with.